### ISSUE TITLE
constfrac Method Giving Error

### ISSUE THREAD

**Issue by acschick** [SeanMcGrath/fadc-tools#5]
When trying to use the constfrac peak detection method on file 311.root, it throws an error as shown below. Other peak detection methods, such as average, still work properly. This is using the file 311.root that I sent you for the previous issue.

In[1]: fadc -p constfrac analyze 7 311.root
Out[1]: terminate called after throwing an instance of 'std::out_of_range'
               what():    vector::_M_range_check
            Aborted (core dumped)

**Comment by SeanMcGrath** [comment]
I can't reproduce this. I was getting this issue earlier but pushed a fix this morning, have you updated?

**Comment by acschick** [comment]
Yes, I even just checked again to make sure. Even after updating, the same error message is appearing, and it is specifically for the constfrac peak detection on channel 7 of 311.root. I have tested some of the other channels, and the error is also given with channels 3, 5, 6, and 8. Other channels appear to be working fine.

**Comment by SeanMcGrath** [comment]
Found the issue -- vectors were being declared at the wrong point in a loop and weren't getting deallocated. This makes the system run out of memory when it tries to handle larger files such as 311.root.

I have pushed a fixed version, which now has a functioning 'analyze all' command.
